spec.bs

```diff
@@ -2921,6 +2926,188 @@ This specification defines two [=policy-controlled features=] identified by the
+new Request (<var ignore>input</var>, <var ignore>init</var>)</a></code> constructor steps, before
+step "Set this's [=Request/request=] to |request|":
+
+1. If <var ignore>init</var>["{{RequestInit/adAuctionHeaders}}"] [=map/exists=], then set
+  |request|'s [=request/capture-ad-auction-headers=] to it.
+
+</div>
+
+<div algorithm="fetch Sec-Ad-Auction-Fetch patch">
+The following step will be added to the [=http-network-or-cache fetch=] algorithm, before step
+"Modify |httpRequest|'s [=request/header list=] per HTTP. ...":
+
+1. If [=request=]'s [=request/capture-ad-auction-headers=] is true, then [=header list/set a
+  structured field value=] given «[:Sec-Ad-Auction-Fetch:], the [=structured header/boolean=] `?1`»
+  in |httpRequest|'s [=header list=].
+
+</div>
+
+<div algorithm="fetch per traversable navigable structures patch">
+Modify [[FETCH]]'s [[FETCH#infrastructure]] to add a new section called "Per Traversable Navigable
+Structures", with the following content:
+
+<h3 id=direct-from-sellers-signals-key-struct>Direct from seller signals key</h3>
+A <dfn>direct from seller signals key</dfn> is a [=struct=] with the following [=struct/items=]:
+
+NOTE: This is only captured during a {{WindowOrWorkerGlobalScope/fetch()}} call with the
+`{adAuctionHeaders: true}` option.
+
+<dl dfn-for="direct from seller signals key">
+: <dfn>seller</dfn>
+:: An [=origin=]. Matches the origin that served the captured [:Ad-Auction-Signals:] header.
+: <dfn>ad slot</dfn>
+:: A [=string=]. Matches the `adSlot` key of the JSON dictionaries in the top-level array of the
+  [:Ad-Auction-Signals:] value.
+
+</dl>
+
+<h3 id=direct-from-sellers-signals-struct>Direct from seller signals</h3>
+A <dfn>direct from seller signals</dfn> is a [=struct=] with the following [=struct/items=]:
+
+NOTE: This is only captured during a {{WindowOrWorkerGlobalScope/fetch()}} call with the
+`{adAuctionHeaders: true}` option.
+
+<dl dfn-for="direct from seller signals">
+: <dfn>seller signals</dfn>
+:: Null or a [=string=].
+  Opaque JSON data passed to both buyers' and the seller's [=script runners=].
+: <dfn>auction signals</dfn>
+:: Null or a [=string=].
+  Opaque JSON data passed to the seller's [=script runner=].
+: <dfn>per buyer signals</dfn>
+:: A [=map=] whose [=map/keys=] are [=origins=] and whose [=map/values=] are [=strings=].
+  [=map/Keys=] are buyers and must be valid HTTPS origins. [=map/Values=] are opaque JSON data
+  passed to corresponding buyer's [=script runner=].
+
+</dl>
+
+Each [=traversable navigable=] has a <dfn>captured ad auction headers</dfn>, which is a [=map=]
+whose [=map/keys=] are [=direct from seller signals keys=] and whose [=map/values=] are
+[=direct from seller signals=].
+</div>
+
+<div algorithm="fetch auction headers patch">
+The following will be added to [[Fetch#http-extensions]]:
+
+<h3 id=sec-ad-auction-fetch-header>`Sec-Ad-Auction-Fetch` header</h3>

```

Comment:
Can we make this heading text actually link to the HTTP header dfn? See https://wicg.github.io/fenced-frame/#supports-loading-mode for example.

Reply:
> So, in your example, the heading doesn't have a link to the dfn, but rather the heading has the dfn itself

No, it has a link to the HTTP header, that itself is defined in another spec. [The code is actually like this](https://github.com/WICG/fenced-frame/blob/master/spec.bs#L2100):

```
<h4 id=supports-loading-mode>The \`<a http-header><code>Supports-Loading-Mode</code></a>\` HTTP response header</h4>
```

Over in https://github.com/WICG/turtledove/pull/771#discussion_r1315154834, I recommended not defining new things in an HTML header. Linking to them is OK IMO. That's very orthogonal though. If I were king of the world, sure I'd make the HTML header https://wicg.github.io/nav-speculation/prerendering.html#http-headerdef-supports-loading-mode _not_ include the definition of the `Supports-Loading-Mode`, and I'd define that HTTP header in a normal paragraph like you do in https://github.com/WICG/turtledove/pull/771/files#diff-6f5a1d8263b0b0c42e2716ba5750e3652e359532647ac934c1c70086ae3ceddaR3048.

Does that make sense?

Reply:
Huh, that makes more sense -- I wonder if I was looking at an old version that had a dfn instead of an <a> tag in the header? When I look now, I see the version with the <a> tag. Weird.

(I was also seeing weird behavior with your links to the rendered spec -- the browser would scroll a bit after navigating, so it was difficult to see what was the destination of the link).

Anyways, done.